src/reporting/bit_list.rs

```diff
@@ -0,0 +1,74 @@
+use std::{u64, convert::TryInto};
+
+pub struct BitList {
+  pub bits: Vec<bool>
+}
+
+impl BitList {
+
+  pub fn concat(&self, bit_list: BitList) -> BitList {
+    BitList { bits: self.bits.clone().into_iter().chain(bit_list.bits.into_iter()).collect() }
```

Comment:
Can extend be used here: https://doc.rust-lang.org/std/iter/trait.Extend.html
(instead of chain/collect)

Reply:
`extend` appears to mutate the vector. Unless we've good reasons I'd prefer to keep immutable/flow style for now. I'd be interested in something to make this shorter though. If there's nothing out of the box, writing our own extensions. We have to investigate.

Reply:
P.S. Given Rust's safety around mutability, I'm open to use it. I find immutability easier to reason about / write in general: e.g. `return old_vec + vec` instead of `old_vec.add(vec); return old_vec`. Maybe we also can write extensions that use mutability internally, with an immutable interface.